spec/models/oauth_token_spec.rb

```diff
@@ -0,0 +1,311 @@
+          end
+        end
+
+      end
+
+      describe "attempt exchange with out authorization" do
+
+        before(:each) do
+          @value = @token.exchange!
+        end
+
+        it "should return false" do
+          expect(@value).to eq false
+        end
+
+        it "should not invalidate request token" do
+          expect(@token).not_to be_invalidated
+        end
+      end
+
+      it "should return 1.0a style to_query" do
+        expect(@token.to_query).to eq "oauth_token=#{@token.token}&oauth_token_secret=#{@token.secret}&oauth_callback_confirmed=true"
+      end
+
+    end
+
+    describe "with oob callback" do
+      before(:each) do
+        @token.callback_url='oob'
+      end
+
+      it "should not be oauth10" do
+        expect(@token).not_to be_oauth10
+      end
+
+      it "should be oob" do
+        expect(@token).to be_oob
+      end
+
+      describe "authorize request" do
+        before(:each) do
+          @token.authorize!(@user)
+        end
+
+        it "should be authorized" do
+          expect(@token).to be_authorized
+        end
+
+        it "should have authorized at" do
+          expect(@token.authorized_at).not_to be_nil
+        end
+
+        it "should have user set" do
+          expect(@token.user).to eq @user
+        end
+
+        it "should have verifier" do
+          expect(@token.verifier).not_to be_nil
+        end
+
+        describe "exchange for access token" do
+
+          before(:each) do
+            @token.provided_oauth_verifier=@token.verifier
+            @access = @token.exchange!
+          end
+
+          it "should invalidate request token" do
+            expect(@token).to be_invalidated
+          end
+
+          it "should set user on access token" do
+            expect(@access.user).to eq @user
+          end
+
+          it "should authorize accesstoken" do
+            expect(@access).to be_authorized
+          end
+        end
+
+        describe "attempt exchange with invalid verifier (OAuth 1.0a)" do
+
+          before(:each) do
+            @value = @token.exchange!
+          end
+
+          it "should return false" do
+            expect(@value).to eq false
+          end
+
+          it "should not invalidate request token" do
+            expect(@token).not_to be_invalidated
+          end
+        end
+
+      end
+
+      describe "attempt exchange with out authorization invalid verifier" do
+
+        before(:each) do
+          @value = @token.exchange!
+        end
+
+        it "should return false" do
+          expect(@value).to eq false
+        end
+
+        it "should not invalidate request token" do
+          expect(@token).not_to be_invalidated
+        end
+      end
+
+      it "should return 1.0 style to_query" do
+        expect(@token.to_query).to eq "oauth_token=#{@token.token}&oauth_token_secret=#{@token.secret}&oauth_callback_confirmed=true"
+      end
+    end
+  end
+
+  if defined? OAUTH_10_SUPPORT && OAUTH_10_SUPPORT
```

Comment:
Do we need OAuth 1 support? I don't expect so, and that might be a big simplification to this PR (I see a lot of chunks that are OAuth 1 dedicated)